Components/LearningCore/Source/gov/sandia/cognition/learning/algorithm/clustering/OptimizedAffinityPropagation.java

```diff
@@ -0,0 +1,261 @@
+package gov.sandia.cognition.learning.algorithm.clustering;

```

Comment:
In your email to me you said this optimized class gives identical results to the original class.  Can you please update the tests for the original class so that this optimized class is tested in the exact same ways?  That way, if a future coder changes something in this class, we can ensure there are no unexpected regressions?

Reply:
Yes for sure. I added a new Test, named OptimizedAffinityPropagationTest, testing the class OptimizedAffinityPropagation exactly in the way the test AffinityPropagationTest tested the class AffinityPropagation. Maybe i should add a new test method in order to test the case in which the user does not specify the self divergence in input. Could be useful for you?